bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/services/embed/OEmbedClientImpl.java

```diff
@@ -158,18 +159,21 @@ protected OEmbedClientImplConfigurationFactory.Config getConfiguration(String ur
     protected InputStream getData(String url) throws IOException, IllegalArgumentException {
         RequestConfig rc = RequestConfig.custom().setConnectTimeout(connectionTimeout).setSocketTimeout(soTimeout)
             .build();
-        HttpClient httpClient;
+        HttpResponse response;
         if (httpClientBuilderFactory != null
                 && httpClientBuilderFactory.newBuilder() != null) {
-            httpClient = httpClientBuilderFactory.newBuilder()
+            try (CloseableHttpClient httpClient = httpClientBuilderFactory.newBuilder()
                     .setDefaultRequestConfig(rc)
-                    .build();
+                    .build()){
+                response = httpClient.execute(new HttpGet(url));
+            }
         } else {
-            httpClient = HttpClients.custom()
+            try (CloseableHttpClient httpClient = HttpClients.custom()
                     .setDefaultRequestConfig(rc)
-                    .build();
+                    .build()){
+                response = httpClient.execute(new HttpGet(url));
+            }
         }
-        HttpResponse response = httpClient.execute(new HttpGet(url));
         return response.getEntity().getContent();
```

Comment:
AFAIK accessing the response's input stream is no longer possible once the surrounding http client is closed.
One rather needs to deserialize it first, preferably via https://www.javadoc.io/doc/org.apache.httpcomponents/httpclient/4.3.2/org/apache/http/impl/client/CloseableHttpClient.html#execute(org.apache.http.HttpHost,%20org.apache.http.HttpRequest,%20org.apache.http.client.ResponseHandler)

Reply:
@kwin , you're right! @andreeadracea , that's also why the ITs are failing:

```
Test Result (7 failures / +6)
    ITHttpTests.com.adobe.cq.wcm.core.components.it.http.ComponentsIT.testEmbed
    E2ESeleniumTestsGroup3.com.adobe.cq.wcm.core.components.it.seljup.tests.embed.v1.EmbedIT.testUrlOEmbedYouTube
    E2ESeleniumTestsGroup3.com.adobe.cq.wcm.core.components.it.seljup.tests.embed.v1.EmbedIT.testUrlOEmbedSoundCloud
    E2ESeleniumTestsGroup3.com.adobe.cq.wcm.core.components.it.seljup.tests.embed.v1.EmbedIT.testUrlValidation
    E2ESeleniumTestsGroup3.com.adobe.cq.wcm.core.components.it.seljup.tests.embed.v2.EmbedIT.testUrlOEmbedYouTube
    E2ESeleniumTestsGroup3.com.adobe.cq.wcm.core.components.it.seljup.tests.embed.v2.EmbedIT.testUrlOEmbedSoundCloud
    E2ESeleniumTestsGroup3.com.adobe.cq.wcm.core.components.it.seljup.tests.embed.v2.EmbedIT.testUrlValidation
```

See a proposed alternative in #2717